### DISCUSSION TITLE
Layout Cards - Load Images

### DISCUSSION THREAD

**Discussion by hugopvl** [directus/directus#8212] [Archive]
Good afternoon people,

Has anyone had this problem ?

Sometimes it doesn't load the images, and it doesn't seem to have a logical sequence.

Thanks.

https://user-images.githubusercontent.com/81326296/134210505-ee4cb587-13f7-4b67-81d4-ef5b7b30c1ed.mp4

**Comment by benhaynes** [comment]
Odd! Anything in the browser's dev tools console log?

**Reply by hugopvl** [reply]
In the logs there is no error information or any warnings, everything is normal.

**Reply by rijkvanzanten** [reply]
@hugopvl The gray box looks like the loading state. If you look at the network tab, could it be that the request for that particular image is still ongoing?

**Reply by hugopvl** [reply]
@rijkvanzanten At the beginning i also thought i might still be loading, but if you wait, since the loading ends. then you browse between collections and loading is fast.

https://user-images.githubusercontent.com/81326296/134217768-f9f02014-3f12-43bc-8dfa-1e6c1afe6ef2.mp4

**Reply by rijkvanzanten** [reply]
> since the loading ends. then you browse between collections and loading is fast.

The first time a new image is being loaded as a thumbnail, the thumbnail is being generated. The second time, the previously generated thumbnail is returned immediately, so that part makes sense. Not sure why it doesn't show up on first load though

**Reply by hugopvl** [reply]
Just a question, was there any change between version rc.90 and version rc.93 on layoutcard ? because in the rc.90 version this doesn't happen, and the configuration is the same.

**Reply by rijkvanzanten** [reply]
There's a ton of changes every release, so that's very likely yeah

**Reply by hugopvl** [reply]
@rijkvanzanten it's better to open an Issue ?

**Reply by rijkvanzanten** [reply]
Hmm normally i'd say absolutely yes, but I'm struggling to think of a way to reproduce this accurately locally 🤔 I haven't seen this issue locally, or on my own production instances, so I'm wondering if there could be a problem in the network setup of this project specifically. Are you observing the same issue locally?

**Reply by hugopvl** [reply]
@rijkvanzanten It's strange, I have two VM's with debian, in a cluster, 4GB of memory, 4 processors per VM, SSD disks, 10GB network card, connected to a NAS with 10GB fiber, which is where I keep the images. the settings are exactly the same.

One of the VM's has directus v9.rc90 with +/- 300 records all with photos, if you change the layout to card it works perfectly.

The other has directus v9.rc93 with 8 records and this situation happens with the photos on the layout card. 

I've analyzed the logs in detail to see if I could find any information but I didn't find anything unusual.

**Reply by joselcvarela** [reply]
Something similar already happened to me and it was about a misconfigured field on model.
Since we were not able to replicate, my advice is create a new model only with files, create some new records on this model, change view to cards and check if still happening. If it's not, then probably something related to the other model is misconfigured. Otherwise it could be a bug, but we will need a dump of your database and the uploads folder in order to replicate.

**Reply by hugopvl** [reply]
@joselcvarela Olá José, Portugal é mesmo enorme, estamos em todo o lado :) e também sou de Braga, mas olha relativamente ao assunto achas que pode ser algum campo mal configurado ? Não estou a ver o que possa ser, depois de falares ainda pensei que poderia ser um M2M que aqui tinha configurado, apaguei-o para testar mas nada de novo... vou experimentar criar uma coleção nova e testar.

**Reply by joselcvarela** [reply]
Hehe mesmo @hugopvl vou apenas pedir para continuarmos em Inglês porque os meus colegas também te podem ajudar. Caso queiras podemos falar no discord, tambem temos um canal em portugues https://directus.chat

So @hugopvl thought it was a M2M misconfigured but the issue still happened. He said he will try with another model.
When you could, tell us how it went.

**Reply by rijkvanzanten** [reply]
Obrigada! Once we know the problem consistently happens we can immediately fix it 🙂 Just a tad difficult to look into it if we don't know how to get it to happen on our end

**Reply by hugopvl** [reply]
@joselcvarela 
@rijkvanzanten 
Good morning guys,

I think I found the solution, at least for now everything is working beautifully.

What I did was the following: in the layout options, image source, I cleared the selection and reselected the image field, I did this in all collections and solved the problem in all of them.

![image](https://user-images.githubusercontent.com/81326296/134650561-8344610c-e131-470c-919d-c0fd60b7ee23.png)

normally when we select layout card the image (the image field) automatically appears, but it should get a little confused when you have more than one possibility i think, so deselecting and selecting the image solves the problem.

Or as there is more than one possibility (image, ai, psd, pdf) is it rotating?

Ha and sorry for my message above in portuguese, but it's always nice when we find a "neighbor" :)

**Reply by benhaynes** [reply]
No worries, it made me happy to see the Portuguese above... that's why we have an international team/product!

**Reply by joselcvarela** [reply]
@hugopvl if it makes me happy to see people from my country using Directus, imagine from my city 😁

About this
> Or as there is more than one possibility (image, ai, psd, pdf) is it rotating?

It is supposed to stick to your selection.
It seems something went wrong on saving the first time.
Did you have upgraded from another version? It could be something different between versions.

**Reply by hugopvl** [reply]
@joselcvarela 
I updated version rc 92 to rc93,
I did another test, and logged in with another user's account, and had to repeat the same process to solve the problem.

**Reply by joselcvarela** [reply]
An easier way to do that is to go to Settings > Presets and adjust your collection for all users:

![image](https://user-images.githubusercontent.com/14039341/134691032-3501a52d-f468-49aa-8c01-085cce52a95f.png)
